app/controllers/application_controller.rb

```diff
@@ -89,7 +89,16 @@ def handle_unverified_request
     # is raised here, but is denied by disable_api.
     #
     # See http://stackoverflow.com/a/15350123 for more information on login CSRF.
-    render_error status: 422, message: 'Invalid form authenticity token.' unless api_request?
+    unless api_request?
+
+      # Check whether user have cookies enabled, otherwise they'll only be
+      # greeted with the CSRF error upon login.
+      cookie_missing = request.cookies['_open_project_session'].nil?
```

Comment:
I think this would be a good time to move the cookie name into the configuration just like it is done for the autologin_cookie_name. Otherwise we have two places now one has to change when the name is to be changed: This file here and `config/initializers/session_store.rb` https://github.com/opf/openproject/blob/dev/config/initializers/session_store.rb#L38